dao/chat.go

```diff
@@ -199,3 +180,30 @@ func (d chatDao) CloseActivePoll(streamID uint) error {
 func (d chatDao) GetChatsByUser(userID uint) (chats []model.Chat, err error) {
 	return chats, d.db.Find(&chats, "user_id = ?", userID).Error
 }
+
+// GetChat returns a chat message with the given id
+func (d chatDao) GetChat(id uint, userID uint) (*model.Chat, error) {
+	var chat *model.Chat
+
+	err := d.db.Preload("Replies").Preload("UserLikes").Preload("AddressedToUsers").Find(&chat, "id = ?", id).Error
+	if err != nil {
+		return chat, err
+	}
+
+	normalizeChat(chat, userID)
+	return chat, nil
+}
```

Comment:
```suggestion
func (d chatDao) GetChat(id uint, userID uint) (*model.Chat, error) {
	var chat model.Chat

	err := d.db.Preload("Replies").Preload("UserLikes").Preload("AddressedToUsers").Find(&chat, "id = ?", id).Error
	if err != nil {
		return &chat, err
	}

	normalizeChat(&chat, userID)
	return &chat, nil
}
```

Maybe this is more idiomatic? Using the `&` operator on a pointer looks a bit weird to me.

Reply:
Oh I see its double, I am pretty sure I can just leave out the `&`. Good catch! 👍